### DISCUSSION TITLE
Shell future

### DISCUSSION THREAD

**Discussion by hugelgupf** [u-root/u-root#2900] [General]
Hi all,

we currently have several different shells to choose from.

- elvish -- `cmds/core/elvish` -- a fork of https://github.com/elves/elvish with some features removed for size (9.8M compiled, unstripped)
- goshsmall gosh -- `cmds/core/gosh` compiled with -tags=goshsmall -- a https://github.com/mvdan/sh derived shell without tab completion & history (4M compiled, unstripped)
- bubbline gosh -- `cmds/core/gosh` without any tags -- a https://github.com/mvdan/sh derived shell with tab completion & history, but 2MB than goshsmall larger due to using very nice editline terminal UI from github.com/knz/bubbline (6M compiled, unstripped)
- goshliner gosh -- https://github.com/u-root/u-root/pull/2897 introduces a `cmds/core/gosh` compiled with -tags=goshliner, which is 200K larger than goshsmall and uses a very bare bones but functioning editline library to provide tab completion & history (4.2M compiled, unstripped)
- rush -- `cmds/exp/rush` - our old old shell, (2.8M compiled, unstripped)

Which of these should we retain, which should we delete?

I'd like to nominate for deletion:

- elvish
- rush (does anyone need a shell smaller than goshsmall?)
- bubbline gosh

**Comment by rminnich** [comment]
I still need rush b/c at the moment it's the only thing that will run on bare metal.
I can try goshsmall but I'm dubious based on past experience.

**Reply by hugelgupf** [reply]
What'd be necessary to run gosh on bare metal?
